core/src/completed_data_sets_service/max_slot.rs

```diff
@@ -0,0 +1,22 @@
+//! Records the maximum observed slot index.
+
+use {
+    super::{CompletedDataSetHandler, SlotEntries},
+    solana_rpc::max_slots::MaxSlots,
+    std::sync::{atomic::Ordering, Arc},
+};
+
+pub struct MaxSlot {
+    stats: Arc<MaxSlots>,
+}
+
+impl MaxSlot {
```

Comment:
Submodule just for recording the maximum slot seen feels like overkill. This also seems a little slower as we'll be updated the atomic directly as opposed to maintaining a local max and then updating the atomic at the end